pybliometrics/scopus/scopus_search.py

```diff
@@ -215,6 +217,179 @@ def get_eids(self):
+                    
+                    if not imitate_scopus_export:
+                        # Remove potential white spaces
+                        key_author = "".join(key_author.split())
+                
+                key = "".join([key_author, year])
+
+                # Authors
+                authors = ""
+                if author_names:
+                    authors = " and ".join(author_names.split(";"))
+
+                # Pages
+                pages = None
+                page_range = result.pageRange
+                if page_range:
+                    pages = page_range.replace("-", " – ")
+
+                # Affiliation
+                affiliation: str = result.affilname
+                if affiliation:
+                    affiliation = "; ".join(affiliation.split(";"))
+
+                # Author keywords
+                author_keywords: str = result.authkeywords
+                if author_keywords:
+                    author_keywords = "; ".join(author_keywords.split(" | "))
+
+                # All information
+                bib_tex_type = None
+                if (document_type in [type_article, type_review, type_short_survey, type_editorial, type_note, type_letter, type_data_paper, type_erratum, type_conference_review, type_conference_paper, type_retracted, type_none] and aggregation_type == aggregation_type_journal) or (document_type in [type_article, type_review, type_short_survey, type_note] and aggregation_type == aggregation_type_trade_journal) or (document_type == type_article and aggregation_type == aggregation_type_none):
+                    bib_tex_type = bib_tex_type_article
+                if aggregation_type == aggregation_type_conference_proceedings or (document_type == type_conference_paper and aggregation_type in [aggregation_type_book, aggregation_type_none]):
+                    bib_tex_type = bib_tex_type_in_proceedings
+                elif aggregation_type == aggregation_type_book_series or (document_type in [type_book_chapter, type_article, type_editorial] and aggregation_type == aggregation_type_book):
+                    bib_tex_type = bib_tex_type_in_collection
+                elif document_type == type_book and aggregation_type == aggregation_type_book:
+                    bib_tex_type = bib_tex_type_book
+                elif document_type == type_report and aggregation_type == aggregation_type_report:
+                    bib_tex_type = bib_tex_type_techreport
+                if bib_tex_type == None:
+                    raise ValueError(f"Unsupported type | Document type: {document_type} | Aggregation type: {aggregation_type} | DOI: https://doi.org/{result.doi}")
+
+
+                fields = []
+
+                fields = self.add_bibtex_field(fields, "author", authors)
+                fields = self.add_bibtex_field(fields, "title", result.title)
+                fields = self.add_bibtex_field(fields, "date", result.coverDate)
```

Comment:
Can you turn this and the following statements into a loop of tuples?
```
[("author", authors), ("title", result.title), ...]
```

Reply:
Not sure if I recall this correctly, but I might have followed the style of the other existing bibtex export function in pybliometrics.

The problem with putting this in a loop is that there are some fields depending on checks. So the list would then need to be constructed in a similar manner, not making the code much shorter. Still probable makes sense to eliminate a bit of duplication!